How to deprecate a recipe option (#5020)
conan-io/conan-center-index: docs/faqs.md

We decided that these packages (as long as they match the premises) should list all the settings needed to build, so building from sources will generate the expected binary, but they will **remove `compiler` setting inside the `package_id()` method**. As a consequence, the CI will generate packages only for one compiler reducing the workload in the pipeline and the number of possible package IDs.

Note about `build_type`.- We retain the `build_type` setting to make it possible for the users to _debug_ these installer packages. We considered removing this settings and it would be possible to compile these packages in _debug_ mode, but if we remove it from the packageID, the compiled package would override the existing _release_ binary, and it'd be quite inconvenient for the users to compile the binary every time they need to switch from _debug_ to _release_.

> ## Can I remove a deprecated option from recipe

Reviewer: It feels a bit circular, with the answer in the question

```suggestion
## Can I remove an option from recipe
```

> No. Changing any option will result in a different package ID and may result a different behavior, the result can break users.

Reviewer 2: please remove that argument, as adding `del self.info.options.foobar` causes the same effect as removing option (different package ID), so there is no advantage for that case

> No. Changing any option will result in a different package ID and may result a different behavior, the result can break users.

Author: ```suggestion
No. Removing any option will result in a missing value, it can break any user who is using it.
```

We decided that these packages (as long as they match the premises) should list all the settings needed to build, so building from sources will generate the expected binary, but they will **remove `compiler` setting inside the `package_id()` method**. As a consequence, the CI will generate packages only for one compiler reducing the workload in the pipeline and the number of possible package IDs.

Note about `build_type`.- We retain the `build_type` setting to make it possible for the users to _debug_ these installer packages. We considered removing this settings and it would be possible to compile these packages in _debug_ mode, but if we remove it from the packageID, the compiled package would override the existing _release_ binary, and it'd be quite inconvenient for the users to compile the binary every time they need to switch from _debug_ to _release_.

## Can I remove an option from recipe

> It's preferable to don't doing it, because it may break other packages which require those deleted options.

Reviewer: ```suggestion
It's preferable to leave all options (ie. not remove them) because it may break other packages which require those deleted options.
```
